flink-runtime/src/main/java/org/apache/flink/runtime/webmonitor/WebMonitorEndpoint.java

```diff
@@ -180,7 +183,7 @@ public WebMonitorEndpoint(
 		this.restConfiguration = Preconditions.checkNotNull(restConfiguration);
 		this.resourceManagerRetriever = Preconditions.checkNotNull(resourceManagerRetriever);
 		this.transientBlobService = Preconditions.checkNotNull(transientBlobService);
-		this.executor = Preconditions.checkNotNull(executor);
+		this.executor = Executors.newFixedThreadPool(endpointConfiguration.getNumThreads(), new ExecutorThreadFactory("Flink-" + getClass().getSimpleName()));
```

Comment:
Could we still keep it like before that we pass in an `executor`? This makes testing much easier because you have control over the actual `ExecutorService` which you can pass in. What we would need is a kind of factory which is used by the `ClusterEntrypoint` to create the `ExecutorService` and the `WebMonitorEndpoint`. What do you think?

Reply:
Who would then be responsible for shutting it down?

Reply:
Given that no test was making use of that feature I'd be inclined to keep the PR as is. If all our tests so far could be implemented without using this feature, what test could possibly require it?

Reply:
It would be the `WebMonitorEndpoint` task to do so. The contract would be that the `WebMonitorEndpoint` is responsible for managing this executor.

In general, it is a good idea to not create things in the constructor of a component, because it makes it really hard to reuse it in a different environment. Instead, it is imo better to pass in instances of some interface. That way, we don't couple components.

Even if we don't do it right now, we should not remove functionality which allowed it in the first place.